client/src/Components/Layouts/SingleCustomer/index.jsx

```diff
@@ -76,32 +83,46 @@ class Profile extends Component {
       });
   }
 
-  convertToObjectForTable = (results) => {
+  convertToObjectForTable = results => {
     const table = results.map(result => {
       const obj = {};
       const key = Object.keys(result)[0];
       obj.key = key;
-      obj.date= result[key][0].date.split('T')[0];
-      obj.status= result[key][0].status;
-      obj.captain= result[key][0].name;
-      obj.price= result[key][0].total + '$';
+      obj.date = result[key][0].date.split("T")[0];
+      obj.status = result[key][0].status;
+      obj.captain = result[key][0].name;
+      obj.price = result[key][0].total + "$";
       obj.place = result[key][0].place_name;
-      obj.items= result[key][0].items_names;
+      obj.items = result[key][0].items_names;
       return obj;
     });
-    this.setState({tableInfo: table});
-  }
-  handleClick = () => {
-    this.setState({ visible: true });
+    this.setState({ tableInfo: table });
+  };
+  deleteRow = id => {
+    this.setState(prev => {
+      return { tableInfo: prev.tableInfo.filter(data => data.key !== id) };
```

Comment:
@engshorouq You don't need to use the updater function here inside setState, you could just access this.state directly:
`this.setState({ tableInfo: this.state.tableInfo.filter( .....`